src/main/java/com/zendesk/maxwell/replication/BinlogConnectorReplicator.java

```diff
@@ -0,0 +1,313 @@
+package com.zendesk.maxwell.replication;
+
+import com.github.shyiko.mysql.binlog.BinaryLogClient;
+import com.github.shyiko.mysql.binlog.event.*;
+import com.github.shyiko.mysql.binlog.event.deserialization.EventDeserializer;
+import com.zendesk.maxwell.MaxwellContext;
+import com.zendesk.maxwell.MaxwellFilter;
+import com.zendesk.maxwell.MaxwellMysqlConfig;
+import com.zendesk.maxwell.bootstrap.AbstractBootstrapper;
+import com.zendesk.maxwell.producer.AbstractProducer;
+import com.zendesk.maxwell.row.RowMap;
+import com.zendesk.maxwell.row.RowMapBuffer;
+import com.zendesk.maxwell.schema.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.SQLException;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Pattern;
+
+public class BinlogConnectorReplicator extends AbstractReplicator implements Replicator {
+	private final long MAX_TX_ELEMENTS = 10000;
+	protected SchemaStore schemaStore;
+
+	private MaxwellFilter filter;
+
+	private final LinkedBlockingDeque<BinlogConnectorEvent> queue = new LinkedBlockingDeque<>(20);
+
+	protected BinlogConnectorEventListener binlogEventListener;
+
+	private final String maxwellSchemaDatabaseName;
+
+	private final BinaryLogClient client;
+
+	static final Logger LOGGER = LoggerFactory.getLogger(MaxwellReplicator.class);
+	private final boolean stopOnEOF;
+	private boolean hitEOF = false;
+
+	public BinlogConnectorReplicator(
+		SchemaStore schemaStore,
+		AbstractProducer producer,
+		AbstractBootstrapper bootstrapper,
+		MaxwellMysqlConfig mysqlConfig,
+		Long replicaServerID,
+		boolean shouldHeartbeat,
+		PositionStoreThread positionStoreThread,
+		String maxwellSchemaDatabaseName,
+		BinlogPosition start,
+		boolean stopOnEOF,
+		String clientID
+	) {
+		super(clientID, bootstrapper, positionStoreThread, maxwellSchemaDatabaseName, producer);
+		this.schemaStore = schemaStore;
+
+		this.client = new BinaryLogClient(mysqlConfig.host, mysqlConfig.port, mysqlConfig.user, mysqlConfig.password);
+
+		EventDeserializer eventDeserializer = new EventDeserializer();
+		eventDeserializer.setCompatibilityMode(EventDeserializer.CompatibilityMode.DATE_AND_TIME_AS_LONG_MICRO,
+			EventDeserializer.CompatibilityMode.CHAR_AND_BINARY_AS_BYTE_ARRAY);
+		this.client.setEventDeserializer(eventDeserializer);
+
+		this.binlogEventListener = new BinlogConnectorEventListener(client, queue);
+		this.client.setBlocking(!stopOnEOF);
+		this.client.registerEventListener(binlogEventListener);
+		this.client.setServerId(replicaServerID.intValue());
+		this.client.setBinlogFilename(start.getFile());
+		this.client.setBinlogPosition(start.getOffset());
+
+		/*
+		this.shouldHeartbeat = shouldHeartbeat;
```

Comment:
yeah.  they have the client ping the server instead of having the server ping (heartbeat) the client.  It's ok.